tests/unit/imports/server/generateJsonSchema.ts

```diff
@@ -0,0 +1,416 @@
+      });
+      const collection = await createTestCollection(schema);
+
+      assert.throws(() => collection.insert({ minLength: 'a' }), /Document failed validation/);
+      assert.throws(() => collection.insert({ maxLength: 'abcd' }), /Document failed validation/);
+      assert.throws(() => collection.insert({ fixedLength: 'ab' }), /Document failed validation/);
+      assert.throws(() => collection.insert({ pattern: '123' }), /Document failed validation/);
+      assert.throws(() => collection.insert({ email: 'foo' }), /Document failed validation/);
+      assert.throws(() => collection.insert({ uuid: 'foo' }), /Document failed validation/);
+
+      assert.doesNotThrow(() => collection.insert({ minLength: 'abcdef' }));
+      assert.doesNotThrow(() => collection.insert({ maxLength: 'ab' }));
+      assert.doesNotThrow(() => collection.insert({ fixedLength: 'abc' }));
+      assert.doesNotThrow(() => collection.insert({ pattern: 'abc' }));
+      assert.doesNotThrow(() => collection.insert({ email: 'jolly-roger@deathandmayhem.com' }));
+      assert.doesNotThrow(() => collection.insert({ uuid: 'a0eebc99-9c0b-4ef8-bb6d-6bb9bd380a11' }));
+    });
+
+    it('respects strings with multiple checks', async function () {
+      const schema = z.object({
+        _id: z.string(),
+        string: z.string().min(3).max(5).regex(/^[a-z]+$/),
+      });
+      const collection = await createTestCollection(schema);
+
+      assert.throws(() => collection.insert({ string: 'a' }), /Document failed validation/);
+      assert.throws(() => collection.insert({ string: 'abcdef' }), /Document failed validation/);
+      assert.throws(() => collection.insert({ string: '123' }), /Document failed validation/);
+
+      assert.doesNotThrow(() => collection.insert({ string: 'abcd' }));
+    });
+
+    it('allows strings with multiple regex checks', async function () {
+      const schema = z.object({
+        _id: z.string(),
+        string: z.string().regex(/^[a-z]+$/).regex(/^[a-z]{3}$/),
+      });
+      const collection = await createTestCollection(schema);
+
+      assert.throws(() => collection.insert({ string: 'a' }), /Document failed validation/);
+      assert.throws(() => collection.insert({ string: 'abcd' }), /Document failed validation/);
+
+      assert.doesNotThrow(() => collection.insert({ string: 'abc' }));
+    });
+
+    it('rejects unsupported checks', function () {
+      const schema = z.object({
+        _id: z.string(),
+        string: z.string().startsWith('a').optional(),
+      });
+      assert.throws(() => generateJsonSchema(schema), /Unsupported string check/);
+    });
+  });
+
+  describe('number fields', function () {
+    it('respects checks on numbers', async function () {
+      const schema = z.object({
+        _id: z.string(),
+        min: z.number().gt(3).optional(),
+        max: z.number().lt(3).optional(),
+        exclusiveMin: z.number().gte(3).optional(),
+        exclusiveMax: z.number().lte(3).optional(),
```

Comment:
I think the behaviors tested here are correct for the field definitions, but I think the names for `min` and `exclusiveMin` are backwards, as are the names for `max` and `exclusiveMax`?  e.g. greater-than 3 means the value must be exclusive-minimum 3

Reply:
Yeah, that's right. I think I originally wrote these with `z.number.min(3)` and then changed them when I realized that `.gt`/`.lt` were the only syntax for writing exclusive bounds. Fixed.